src/Snapshotting/Tests/LightSwitch.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace EventSauce\EventSourcing\Snapshotting\Tests;
+
+use EventSauce\EventSourcing\AggregateRootBehaviour;
+use EventSauce\EventSourcing\Snapshotting\SnapshottingBehaviour;
+use EventSauce\EventSourcing\Snapshotting\AggregateRootWithSnapshotting;
+
+class LightSwitch implements AggregateRootWithSnapshotting
+{
+    use AggregateRootBehaviour, SnapshottingBehaviour;
+
+    const OFF = false;
+    const ON = true;
+
+    private $state = self::OFF;
+
+    private function createSnapshotState()
+    {
+        return $this->state;
+    }
+
+    private function setPayloadState($state)
```

Comment:
You could consider adding `void` here, as well as to `applyLightSwitchWasFlipped` below.